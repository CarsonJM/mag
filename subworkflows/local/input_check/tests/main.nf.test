nextflow_workflow {

    name "Test Subworkflow: INPUT_CHECK"
    script "subworkflows/local/input_check/main.nf"
    workflow "INPUT_CHECK"
    tag "subworkflows"
    tag "subworkflows_local"
    tag "input_check"


    test("Parameters: default") {

        when {
            params {
                outdir          = "$outputDir"
                input           = 'https://raw.githubusercontent.com/nf-core/test-datasets/mag/samplesheets/samplesheet.csv'
                assembly_input  = 'https://raw.githubusercontent.com/nf-core/test-datasets/mag/samplesheets/assembly_samplesheet.csv'
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }
    }
}
