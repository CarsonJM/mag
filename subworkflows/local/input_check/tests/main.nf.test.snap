{
    "Parameters: default": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_minigut",
                            "run": "0",
                            "group": "0",
                            "single_end": false
                        },
                        [
                            "/nf-core/test-datasets/raw/mag/test_data/test_minigut_R1.fastq.gz",
                            "/nf-core/test-datasets/raw/mag/test_data/test_minigut_R2.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "id": "test_minigut_sample2",
                            "run": "0",
                            "group": "0",
                            "single_end": false
                        },
                        [
                            "/nf-core/test-datasets/raw/mag/test_data/test_minigut_sample2_R1.fastq.gz",
                            "/nf-core/test-datasets/raw/mag/test_data/test_minigut_sample2_R2.fastq.gz"
                        ]
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    [
                        {
                            "id": "test_minigut",
                            "group": "0",
                            "assembler": "MEGAHIT"
                        },
                        [
                            "/nf-core/test-datasets/raw/mag/assemblies/MEGAHIT-test_minigut.contigs.fa.gz"
                        ]
                    ],
                    [
                        {
                            "id": "test_minigut",
                            "group": "0",
                            "assembler": "SPAdes"
                        },
                        [
                            "/nf-core/test-datasets/raw/mag/assemblies/SPAdes-test_minigut_contigs.fasta.gz"
                        ]
                    ],
                    [
                        {
                            "id": "test_minigut_sample2",
                            "group": "0",
                            "assembler": "MEGAHIT"
                        },
                        [
                            "/nf-core/test-datasets/raw/mag/assemblies/MEGAHIT-test_minigut_sample2.contigs.fa.gz"
                        ]
                    ],
                    [
                        {
                            "id": "test_minigut_sample2",
                            "group": "0",
                            "assembler": "SPAdes"
                        },
                        [
                            "/nf-core/test-datasets/raw/mag/assemblies/SPAdes-test_minigut_sample2_contigs.fasta.gz"
                        ]
                    ]
                ],
                "input_assemblies": [
                    [
                        {
                            "id": "test_minigut",
                            "group": "0",
                            "assembler": "MEGAHIT"
                        },
                        [
                            "/nf-core/test-datasets/raw/mag/assemblies/MEGAHIT-test_minigut.contigs.fa.gz"
                        ]
                    ],
                    [
                        {
                            "id": "test_minigut",
                            "group": "0",
                            "assembler": "SPAdes"
                        },
                        [
                            "/nf-core/test-datasets/raw/mag/assemblies/SPAdes-test_minigut_contigs.fasta.gz"
                        ]
                    ],
                    [
                        {
                            "id": "test_minigut_sample2",
                            "group": "0",
                            "assembler": "MEGAHIT"
                        },
                        [
                            "/nf-core/test-datasets/raw/mag/assemblies/MEGAHIT-test_minigut_sample2.contigs.fa.gz"
                        ]
                    ],
                    [
                        {
                            "id": "test_minigut_sample2",
                            "group": "0",
                            "assembler": "SPAdes"
                        },
                        [
                            "/nf-core/test-datasets/raw/mag/assemblies/SPAdes-test_minigut_sample2_contigs.fasta.gz"
                        ]
                    ]
                ],
                "raw_long_reads": [
                    
                ],
                "raw_short_reads": [
                    [
                        {
                            "id": "test_minigut",
                            "run": "0",
                            "group": "0",
                            "single_end": false
                        },
                        [
                            "/nf-core/test-datasets/raw/mag/test_data/test_minigut_R1.fastq.gz",
                            "/nf-core/test-datasets/raw/mag/test_data/test_minigut_R2.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "id": "test_minigut_sample2",
                            "run": "0",
                            "group": "0",
                            "single_end": false
                        },
                        [
                            "/nf-core/test-datasets/raw/mag/test_data/test_minigut_sample2_R1.fastq.gz",
                            "/nf-core/test-datasets/raw/mag/test_data/test_minigut_sample2_R2.fastq.gz"
                        ]
                    ]
                ]
            }
        ],
        "timestamp": "2023-11-29T19:25:35+0000"
    }
}